src/drivers/nuopc/mediator/med.F90

```diff
@@ -513,238 +511,219 @@ subroutine InitializeIPDv03p1(gcomp, importState, exportState, clock, rc)
     ! actually set them to anything
     !------------------
 
-    call shr_nuopc_fldList_Zero(fldsFrAtm, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsToAtm, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsFrOcn, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsToOcn, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsFrIce, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsToIce, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsFrLnd, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsToLnd, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsFrRof, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsToRof, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsFrWav, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsToWav, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsFrGlc, rc=rc)
-    call shr_nuopc_fldList_Zero(fldsToGlc, rc=rc)
+    do n1 = 1,ncomps
+      call shr_nuopc_fldList_Zero(fldsFr(n1), rc=rc)
+      call shr_nuopc_fldList_Zero(fldsTo(n1), rc=rc)
+    enddo
     call shr_nuopc_fldList_Zero(fldsAtmOcn,rc=rc)
 
     !------------------
-    ! Create fldsToAtm
+    ! Create fldsTo(compatm)
```

Comment:
These set of lines should be encapsulated into one routine and not have a set of repeated calls.
e.g. 

call shr_nuopc_fldlist_init_seqflds(fldsTo=fldsTo(compatm), fldlistTo=seq_flds_x2a_states, mappingTo='bilinear', fldsFrom=fldsFrom(compatm), fldlistFrom=seq_flds_a2x_states, mappingFrom='conservative')

Or something like that.

Reply:
For now, this code is largely a place holder until we develop a better way to define the coupling fields and associated a mapping approach/file to each field.  We could encapsulate this chunk of code, but we also have some more design work to do with regard to; defining fields, potentially specifying the mapping approach on a field by field basis, defining the mapping files and/or support for on the fly weights generation, and ultimately how all these things interact.